cmake can't find gtest.lib on repeated runs of ament test on Windows

`ament.py test` can build and run gtests on an empty workspace. However after a workspace has been populated, any subsequent runs will fail to find `gtest.lib`:

```
11>InitializeBuildStatus:
         Creating "ament_index_cpp_utest.dir\Debug\ament_in.4C53B27F.tlog\unsuc
         cessfulbuild" because "AlwaysCreate" was specified.
       CustomBuild:
         All outputs are up-to-date.
       ClCompile:
         All outputs are up-to-date.
       Link:
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\link.
         exe /ERRORREPORT:QUEUE /OUT:"C:\dev\ros2\build\ament_index_cpp\Debug\a
         ment_index_cpp_utest.exe" /INCREMENTAL /NOLOGO kernel32.lib user32.lib
          gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib co
         mdlg32.lib advapi32.lib gtest.lib gtest_main.lib Debug\ament_index_cpp
         .lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /mani
         fest:embed /Debug /PDB:"C:/dev/ros2/build/ament_index_cpp/Debug/ament_
         index_cpp_utest.pdb" /SUBSYSTEM:CONSOLE /TLBID:1 /DYNAMICBASE /NXCOMPA
         T /IMPLIB:"C:/dev/ros2/build/ament_index_cpp/Debug/ament_index_cpp_ute
         st.lib" /MACHINE:X64  /machine:x64 /debug ament_index_cpp_utest.dir\De
         bug\utest.obj
    11>LINK : fatal error LNK1104: cannot open file 'gtest.lib' [C:\dev\ros2\bu
       ild\ament_index_cpp\ament_index_cpp_utest.vcxproj]
    11>Done Building Project "C:\dev\ros2\build\ament_index_cpp\ament_index_cpp
       _utest.vcxproj" (default targets) -- FAILED.
     4>Done Building Project "C:\dev\ros2\build\ament_index_cpp\ament_index_cpp
       _utest.vcxproj.metaproj" (default targets) -- FAILED.
     2>Done Building Project "C:\dev\ros2\build\ament_index_cpp\ALL_BUILD.vcxpr
       oj.metaproj" (default targets) -- FAILED.
     1>Done Building Project "C:\dev\ros2\build\ament_index_cpp\ament_index_cpp
       .sln" (default targets) -- FAILED.

Build FAILED.

       "C:\dev\ros2\build\ament_index_cpp\ament_index_cpp.sln" (default target)
        (1) ->
       "C:\dev\ros2\build\ament_index_cpp\ament_index_cpp_utest.vcxproj.metapro
       j" (default target) (4) ->
       "C:\dev\ros2\build\ament_index_cpp\ament_index_cpp_utest.vcxproj" (defau
       lt target) (11) ->
       (Link target) ->
         LINK : fatal error LNK1104: cannot open file 'gtest.lib' [C:\dev\ros2\
       build\ament_index_cpp\ament_index_cpp_utest.vcxproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:03.56

<== Command 'C:\dev\ros2\build\ament_index_cpp\cmake__build.bat C:\Program Files
 (x86)\MSBuild\14.0\bin\amd64\msbuild.EXE /m C:\dev\ros2\build\ament_index_cpp\a
ment_index_cpp.sln' failed with exit code '1'

C:\dev\ros2>
```

--- a/ament_cmake_gmock/ament_cmake_gmock-extras.cmake
+++ b/ament_cmake_gmock/ament_cmake_gmock-extras.cmake
@@ -1,4 +1,4 @@
-# Copyright 2014 Open Source Robotics Foundation, Inc.
+# Copyright 2014-2015 Open Source Robotics Foundation, Inc.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -22,18 +22,22 @@ macro(_ament_cmake_gmock_find_gmock)
     find_package(ament_cmake_test QUIET REQUIRED)
     find_package(gmock_vendor QUIET)
 
-    if(NOT GMOCK_FOUND)
+    # if gmock sources were not found in a previous run
+    if(NOT GMOCK_FROM_SOURCE_FOUND)
       # search path for gmock includes and sources
       set(_search_path_include "")
       set(_search_path_src "")
+
       # option() consider environment variable to find gmock
       if(NOT "$ENV{GMOCK_DIR} " STREQUAL " ")
         list(APPEND _search_path_include "$ENV{GMOCK_DIR}/include/gmock")
         list(APPEND _search_path_src "$ENV{GMOCK_DIR}/src")
       endif()
+
       # check to system installed path (i.e. on Ubuntu)
       set(_search_path_include "/usr/include/gmock")
       set(_search_path_src "/usr/src/gmock/src")
+
       # check gmock_vendor path
       if(gmock_vendor_FOUND AND gmock_vendor_BASE_DIR)
         list(APPEND _search_path_include "${gmock_vendor_BASE_DIR}/include/gmock")
@@ -52,84 +56,65 @@ macro(_ament_cmake_gmock_find_gmock)
       )
 
       if(_gmock_header_file AND _gmock_src_file)
-        get_filename_component(_gmock_base_dir "${_gmock_src_file}" PATH)
-        get_filename_component(_gmock_base_dir "${_gmock_base_dir}" PATH)
-        # add CMakeLists.txt from gmock dir
-        set(_gmock_binary_dir "${CMAKE_BINARY_DIR}/gmock")
-        add_subdirectory("${_gmock_base_dir}" ${_gmock_binary_dir})
-        # mark gmock targets with EXCLUDE_FROM_ALL to only build
-        # when tests are built which depend on them
-        set_target_properties(gmock gmock_main PROPERTIES EXCLUDE_FROM_ALL 1)
-        get_filename_component(_gmock_include_dir "${_gmock_header_file}" PATH)
-        get_filename_component(_gmock_include_dir ${_gmock_include_dir} PATH)
         # set from-source variables
         set(GMOCK_FROM_SOURCE_FOUND TRUE CACHE INTERNAL "")
-        set(GMOCK_FROM_SOURCE_INCLUDE_DIRS "${_gmock_include_dir}"
-          CACHE INTERNAL "")
-        set(GMOCK_FROM_SOURCE_LIBRARY_DIRS "${_gmock_binary_dir}"
-          CACHE INTERNAL "")
+
+        get_filename_component(_gmock_base_dir "${_gmock_src_file}" PATH)
+        get_filename_component(_gmock_base_dir "${_gmock_base_dir}" PATH)
+        set(GMOCK_FROM_SOURCE_BASE_DIR "${_gmock_base_dir}" CACHE INTERNAL "")
+
+        get_filename_component(_gmock_include_dir "${_gmock_header_file}" PATH)
+        get_filename_component(_gmock_include_dir ${_gmock_include_dir} PATH)
+        set(GMOCK_FROM_SOURCE_INCLUDE_DIRS "${_gmock_include_dir}" CACHE INTERNAL "")
+
+        set(GMOCK_FROM_SOURCE_LIBRARY_DIRS "${CMAKE_BINARY_DIR}/gmock" CACHE INTERNAL "")
+
         set(GMOCK_FROM_SOURCE_LIBRARIES "gmock" CACHE INTERNAL "")
         set(GMOCK_FROM_SOURCE_MAIN_LIBRARIES "gmock_main" CACHE INTERNAL "")
-        message(STATUS "Found gmock sources under '${_gmock_base_dir}': "
-          "C++ tests using 'Google Mock' will be built")
-
-        if(NOT "${_gmock_base_dir} " STREQUAL "${gmock_vendor_BASE_DIR} ")
-          set(_gtest_base_dir "${_gmock_base_dir}/gtest")
-          # set from-source variables for embedded gtest
-          set(GTEST_FROM_SOURCE_FOUND TRUE CACHE INTERNAL "")
-          set(GTEST_FROM_SOURCE_INCLUDE_DIRS "${_gtest_base_dir}/include"
-            CACHE INTERNAL "")
-          set(GTEST_FROM_SOURCE_LIBRARY_DIRS "${_gmock_binary_dir}/gtest"
-            CACHE INTERNAL "")
-          set(GTEST_FROM_SOURCE_LIBRARIES "gtest" CACHE INTERNAL "")
-          set(GTEST_FROM_SOURCE_MAIN_LIBRARIES "gtest_main" CACHE INTERNAL "")
-          message(STATUS "Found gtest sources under '${_gtest_base_dir}': "
-            "C++ tests using 'Google Test' will be built")
-        else()
-          # if gmock_vendor is being used it should not be used for gtests
-          # instead gtest_vendor should be used directly
-        endif()
-      endif()
-      if(GMOCK_FROM_SOURCE_FOUND)
-        # set the same variables as find_package() would set
-        # but do NOT set GMOCK_FOUND in the cache when using gmock from source
-        # since the subdirectory must always be added to add the gmock targets
-        set(GMOCK_FOUND ${GMOCK_FROM_SOURCE_FOUND})
-        set(GMOCK_INCLUDE_DIRS ${GMOCK_FROM_SOURCE_INCLUDE_DIRS})
-        set(GMOCK_LIBRARY_DIRS ${GMOCK_FROM_SOURCE_LIBRARY_DIRS})
-        set(GMOCK_LIBRARIES ${GMOCK_FROM_SOURCE_LIBRARIES})
-        set(GMOCK_MAIN_LIBRARIES ${GMOCK_FROM_SOURCE_MAIN_LIBRARIES})
-        # also set the gtest variables
-        # but do NOT set GTEST_FOUND in the cache when using gtest from source
-        # since the subdirectory must always be added to add the gmock targets
-        set(GTEST_FOUND ${GTEST_FROM_SOURCE_FOUND})
-        set(GTEST_INCLUDE_DIRS ${GTEST_FROM_SOURCE_INCLUDE_DIRS})
-        set(GTEST_LIBRARY_DIRS ${GTEST_FROM_SOURCE_LIBRARY_DIRS})
-        set(GTEST_LIBRARIES ${GTEST_FROM_SOURCE_LIBRARIES})
-        set(GTEST_MAIN_LIBRARIES ${GTEST_FROM_SOURCE_MAIN_LIBRARIES})
-        set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
-        # gmock adds its own include dir first and then its embedded gtest one
-        # but when gtest is installed beside gmock it will overlay the gtest
-        # headers embedded in gmock
-        # therefore prepending the include dir of the embedded gtest version
-        # to ensure it is being used
-        if(NOT "${GTEST_INCLUDE_DIRS} " STREQUAL " ")
-          include_directories(BEFORE "${GTEST_INCLUDE_DIRS}")
-        endif()
       endif()
     endif()
+
+    if(GMOCK_FROM_SOURCE_FOUND)
+      message(STATUS "Found gmock sources under '${GMOCK_FROM_SOURCE_BASE_DIR}': "
+        "C++ tests using 'Google Mock' will be built")
+
+      # add CMakeLists.txt from gmock dir
+      add_subdirectory("${GMOCK_FROM_SOURCE_BASE_DIR}" "${CMAKE_BINARY_DIR}/gmock")
+
+      # mark gmock targets with EXCLUDE_FROM_ALL to only build
+      # when tests are built which depend on them
+      set_target_properties(gmock gmock_main PROPERTIES EXCLUDE_FROM_ALL 1)
+
+      # set the same variables as find_package() would set
+      # but do NOT set GMOCK_FOUND in the cache when using gmock from source
+      # since the subdirectory must always be added to add the gmock targets
+      set(GMOCK_FOUND ${GMOCK_FROM_SOURCE_FOUND})
+      set(GMOCK_INCLUDE_DIRS ${GMOCK_FROM_SOURCE_INCLUDE_DIRS})
+      set(GMOCK_LIBRARY_DIRS ${GMOCK_FROM_SOURCE_LIBRARY_DIRS})
+      set(GMOCK_LIBRARIES ${GMOCK_FROM_SOURCE_LIBRARIES})
+      set(GMOCK_MAIN_LIBRARIES ${GMOCK_FROM_SOURCE_MAIN_LIBRARIES})
+
+      if("${GMOCK_FROM_SOURCE_BASE_DIR} " STREQUAL "${gmock_vendor_BASE_DIR} ")
+        # if gmock_vendor is being used it should not be used for gtests
+        # instead gtest_vendor should be used directly
+      else()
+        set(_gtest_base_dir "${GMOCK_FROM_SOURCE_BASE_DIR}/gtest")
+        # set from-source variables for embedded gtest
+        set(GTEST_FROM_SOURCE_FOUND TRUE CACHE INTERNAL "")
+        set(GTEST_FROM_SOURCE_BASE_DIR "${_gtest_base_dir}" CACHE INTERNAL "")
+        set(GTEST_FROM_SOURCE_INCLUDE_DIRS "${_gtest_base_dir}/include" CACHE INTERNAL "")
+        set(GTEST_FROM_SOURCE_LIBRARY_DIRS "${_gmock_binary_dir}/gtest" CACHE INTERNAL "")
+        set(GTEST_FROM_SOURCE_LIBRARIES "gtest" CACHE INTERNAL "")
+        set(GTEST_FROM_SOURCE_MAIN_LIBRARIES "gtest_main" CACHE INTERNAL "")
+      endif()
+    endif()
+
     if(NOT GMOCK_FOUND)
       message(WARNING
         "'gmock' not found, C++ tests using 'Google Mock' can not be built. "
         "Please install the 'Google Mock' headers globally in your system to "
         "enable these tests (e.g. on Ubuntu/Debian install the package "
-          "'google-mock')")
-    endif()
-
-    # for Visual 2012 we need to increase the fixed variadic template size to
-    # build gtest (https://code.google.com/p/googletest/source/detail?r=675)
-    if(GMOCK_FOUND AND MSVC AND MSVC_VERSION EQUAL 1700)
-      add_definitions(/D _VARIADIC_MAX=10)
+          "'google-mock') or get the ament package 'gmock_vendor'")
     endif()
   endif()
 endmacro()

--- a/ament_cmake_gtest/ament_cmake_gtest-extras.cmake
+++ b/ament_cmake_gtest/ament_cmake_gtest-extras.cmake
@@ -1,4 +1,4 @@
-# Copyright 2014 Open Source Robotics Foundation, Inc.
+# Copyright 2014-2015 Open Source Robotics Foundation, Inc.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -20,96 +20,101 @@ macro(_ament_cmake_gtest_find_gtest)
     set(_AMENT_CMAKE_GTEST_FIND_GTEST TRUE)
 
     find_package(ament_cmake_test QUIET REQUIRED)
-    find_package(gtest_vendor QUIET)
 
     # allow other packages to find gtest instead
     ament_execute_extensions(ament_cmake_gtest_find_gtest)
 
-    if(NOT GTEST_FOUND)
+    # if gtest sources were not found in a previous run
+    if(NOT GTEST_FROM_SOURCE_FOUND)
+      # try to find built versions of gtest
       find_package(GTest QUIET)
+
+      # otherwise search for sources
+      if(NOT GTEST_FOUND)
+        # search path for gtest includes and sources
+        set(_search_path_include "")
+        set(_search_path_src "")
+
+        # option() consider environment variable to find gtest
+        if(NOT "$ENV{GTEST_DIR} " STREQUAL " ")
+          list(APPEND _search_path_include "$ENV{GTEST_DIR}/include/gtest")
+          list(APPEND _search_path_src "$ENV{GTEST_DIR}/src")
+        endif()
+
+        # check to system installed path (i.e. on Ubuntu)
+        set(_search_path_include "/usr/include/gtest")
+        set(_search_path_src "/usr/src/gtest/src")
+
+        # check gtest_vendor path
+        find_package(gtest_vendor QUIET)
+        if(gtest_vendor_FOUND AND gtest_vendor_BASE_DIR)
+          list(APPEND _search_path_include "${gtest_vendor_BASE_DIR}/include/gtest")
+          list(APPEND _search_path_src "${gtest_vendor_BASE_DIR}/src")
+        endif()
+
+        find_file(_gtest_header_file "gtest.h"
+          PATHS ${_search_path_include}
+          NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH
+        )
+        find_file(_gtest_src_file
+          "gtest.cc"
+          "gtest-all.cc"  # alternative when using "fused" sources
+          PATHS ${_search_path_src}
+          NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH
+        )
+
+        if(_gtest_header_file AND _gtest_src_file)
+          # set from-source variables
+          set(GTEST_FROM_SOURCE_FOUND TRUE CACHE INTERNAL "")
+
+          get_filename_component(_gtest_base_dir "${_gtest_src_file}" PATH)
+          get_filename_component(_gtest_base_dir "${_gtest_base_dir}" PATH)
+          set(GTEST_FROM_SOURCE_BASE_DIR "${_gtest_base_dir}" CACHE INTERNAL "")
+
+          get_filename_component(_gtest_include_dir "${_gtest_header_file}" PATH)
+          get_filename_component(_gtest_include_dir ${_gtest_include_dir} PATH)
+          set(GTEST_FROM_SOURCE_INCLUDE_DIRS "${_gtest_include_dir}" CACHE INTERNAL "")
+
+          set(GTEST_FROM_SOURCE_LIBRARY_DIRS "${CMAKE_BINARY_DIR}/gtest" CACHE INTERNAL "")
+
+          set(GTEST_FROM_SOURCE_LIBRARIES "gtest" CACHE INTERNAL "")
+          set(GTEST_FROM_SOURCE_MAIN_LIBRARIES "gtest_main" CACHE INTERNAL "")
+        endif()
+      endif()
     endif()
-    if(GTEST_FOUND)
-      message(STATUS "Found gtest: C++ tests using 'Google Test' will be built")
-      set(GTEST_FOUND ${GTEST_FOUND} CACHE INTERNAL "")
-      set(GTEST_INCLUDE_DIRS ${GTEST_INCLUDE_DIRS} CACHE INTERNAL "")
-      set(GTEST_LIBRARIES ${GTEST_LIBRARIES} CACHE INTERNAL "")
-      set(GTEST_MAIN_LIBRARIES ${GTEST_MAIN_LIBRARIES} CACHE INTERNAL "")
-      set(GTEST_BOTH_LIBRARIES ${GTEST_BOTH_LIBRARIES} CACHE INTERNAL "")
-    else()
-      # search path for gtest includes and sources
-      set(_search_path_include "")
-      set(_search_path_src "")
-      # option() consider environment variable to find gtest
-      if(NOT "$ENV{GTEST_DIR} " STREQUAL " ")
-        list(APPEND _search_path_include "$ENV{GTEST_DIR}/include/gtest")
-        list(APPEND _search_path_src "$ENV{GTEST_DIR}/src")
-      endif()
-      # check to system installed path (i.e. on Ubuntu)
-      set(_search_path_include "/usr/include/gtest")
-      set(_search_path_src "/usr/src/gtest/src")
-      # check gtest_vendor path
-      if(gtest_vendor_FOUND AND gtest_vendor_BASE_DIR)
-        list(APPEND _search_path_include "${gtest_vendor_BASE_DIR}/include/gtest")
-        list(APPEND _search_path_src "${gtest_vendor_BASE_DIR}/src")
-      endif()
 
-      find_file(_gtest_header_file "gtest.h"
-        PATHS ${_search_path_include}
-        NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH
-      )
-      find_file(_gtest_src_file
-        "gtest.cc"
-        "gtest-all.cc"  # alternative when using "fused" sources
-        PATHS ${_search_path_src}
-        NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH
-      )
+    if(GTEST_FROM_SOURCE_FOUND)
+      message(STATUS "Found gtest sources under '${GTEST_FROM_SOURCE_BASE_DIR}': "
+        "C++ tests using 'Google Test' will be built")
 
-      if(_gtest_header_file AND _gtest_src_file)
-        get_filename_component(_gtest_base_dir "${_gtest_src_file}" PATH)
-        get_filename_component(_gtest_base_dir "${_gtest_base_dir}" PATH)
+      # if the gtest sources are from a subfolder of gmock (not the case for gmock_vendor)
+      # the subdirectory was already added by gmock
+      if(NOT TARGET gtest)
         # add CMakeLists.txt from gtest dir
-        set(_gtest_binary_dir "${CMAKE_BINARY_DIR}/gtest")
-        add_subdirectory("${_gtest_base_dir}" ${_gtest_binary_dir})
+        add_subdirectory("${GTEST_FROM_SOURCE_BASE_DIR}" "${CMAKE_BINARY_DIR}/gtest")
+
         # mark gtest targets with EXCLUDE_FROM_ALL to only build
         # when tests are built which depend on them
         set_target_properties(gtest gtest_main PROPERTIES EXCLUDE_FROM_ALL 1)
-        get_filename_component(_gtest_include_dir "${_gtest_header_file}" PATH)
-        get_filename_component(_gtest_include_dir ${_gtest_include_dir} PATH)
-        # set from-source variables
-        set(GTEST_FROM_SOURCE_FOUND TRUE CACHE INTERNAL "")
-        set(GTEST_FROM_SOURCE_INCLUDE_DIRS "${_gtest_include_dir}"
-          CACHE INTERNAL "")
-        set(GTEST_FROM_SOURCE_LIBRARY_DIRS "${_gtest_binary_dir}"
-          CACHE INTERNAL "")
-        set(GTEST_FROM_SOURCE_LIBRARIES "gtest" CACHE INTERNAL "")
-        set(GTEST_FROM_SOURCE_MAIN_LIBRARIES "gtest_main" CACHE INTERNAL "")
-        message(STATUS "Found gtest sources under '${_gtest_base_dir}': "
-          "C++ tests using 'Google Test' will be built")
       endif()
-      if(GTEST_FROM_SOURCE_FOUND)
-        # set the same variables as find_package()
-        # but do NOT set GTEST_FOUND in the cache when using gtest from source
-        # since the subdirectory must always be added to add the gtest targets
-        set(GTEST_FOUND ${GTEST_FROM_SOURCE_FOUND})
-        set(GTEST_INCLUDE_DIRS ${GTEST_FROM_SOURCE_INCLUDE_DIRS})
-        set(GTEST_LIBRARY_DIRS ${GTEST_FROM_SOURCE_LIBRARY_DIRS})
-        set(GTEST_LIBRARIES ${GTEST_FROM_SOURCE_LIBRARIES})
-        set(GTEST_MAIN_LIBRARIES ${GTEST_FROM_SOURCE_MAIN_LIBRARIES})
-        set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
-      endif()
-      if(NOT GTEST_FOUND)
-        message(WARNING
-          "'gtest' not found, C++ tests using 'Google Test' can not be built. "
-          "Please install the 'Google Test' headers globally in your system "
-          "to enable these tests (e.g. on Ubuntu/Debian install the package "
-          "'libgtest-dev')")
-      endif()
+
+      # set the same variables as find_package()
+      # but do NOT set GTEST_FOUND in the cache when using gtest from source
+      # since the subdirectory must always be added to add the gtest targets
+      set(GTEST_FOUND ${GTEST_FROM_SOURCE_FOUND})
+      set(GTEST_INCLUDE_DIRS ${GTEST_FROM_SOURCE_INCLUDE_DIRS})
+      set(GTEST_LIBRARY_DIRS ${GTEST_FROM_SOURCE_LIBRARY_DIRS})
+      set(GTEST_LIBRARIES ${GTEST_FROM_SOURCE_LIBRARIES})
+      set(GTEST_MAIN_LIBRARIES ${GTEST_FROM_SOURCE_MAIN_LIBRARIES})
+      set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
     endif()
 
-    # for Visual 2012 we need to increase the fixed variadic template size to
-    # build gtest (https://code.google.com/p/googletest/source/detail?r=675)
-    if(GTEST_FOUND AND MSVC AND MSVC_VERSION EQUAL 1700)
-      add_definitions(/D _VARIADIC_MAX=10)
+    if(NOT GTEST_FOUND)
+      message(WARNING
+        "'gtest' not found, C++ tests using 'Google Test' can not be built. "
+        "Please install the 'Google Test' headers globally in your system "
+        "to enable these tests (e.g. on Ubuntu/Debian install the package "
+        "'libgtest-dev') or get the ament package 'gtest_vendor'")
     endif()
   endif()
 endmacro()

